From: David Hildenbrand <david@redhat.com>
Subject: fixup: mm/userfaultfd: enable writenotify while userfaultfd-wp is enabled for a VMA
Date: Sat, 24 Dec 2022 15:02:36 +0100

No need for additional harmless checks if we're wr-protecting either way.

Link: https://lkml.kernel.org/r/71412742-a71f-9c74-865f-773ad83db7a5@redhat.com
Signed-off-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/userfaultfd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/userfaultfd.c~mm-userfaultfd-rely-on-vma-vm_page_prot-in-uffd_wp_range-fix
+++ a/mm/userfaultfd.c
@@ -727,7 +727,7 @@ void uffd_wp_range(struct mm_struct *dst
 	 * to be write-protected as default whenever protection changes.
 	 * Try upgrading write permissions manually.
 	 */
-	if (vma_wants_manual_pte_write_upgrade(dst_vma))
+	if (!enable_wp && vma_wants_manual_pte_write_upgrade(dst_vma))
 		mm_cp_flags |= MM_CP_TRY_CHANGE_WRITABLE;
 	tlb_gather_mmu(&tlb, dst_mm);
 	change_protection(&tlb, dst_vma, start, start + len, vma->vm_page_prot,
_
